### PR TITLE
Profiling changes and data

### PR DESCRIPTION
**Pull request by efischer19**

### CODE REVIEW

**Comment by edx-webhook** [comment]
## PR Cover Letter
- [ ] A detailed description of the changes in the body of the PR
- [ ] If available, a link to the relevant JIRA ticket in the body of the PR
  - `[PROJ-XXXX](https://openedx.atlassian.net/browse/PROJ-XXXX)`

## Sandbox
- [ ] Build a sandbox for your branch and add a link here

## Testing
- [ ] i18n
- [ ] RTL
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

## Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review #1
- [ ] Code review #2
- [ ] Docs review (required for UI strings/error messages)
- [ ] UX review
- [ ] Accessibility review
- [ ] Product review

## Documentation considerations
- [ ] Are there user visible changes? If so, list who is affected, and include a pointer to the relevant code.
- [ ] Should this be mentioned in the release notes? If so, make sure that you included a description.

## DevOps assistance
If it's needed, or even if you suspect it might be needed, do any/all these things as early as possible. Don't surprise devops right before merging! For example, touching envs/aws.py or or envs/common.py are dead giveaways that you'll need assistance when pushing your changes out. Things that require devops attention include, but are not limited to:
- configuration changes
- feature flags
- migrations
Ways to get devops attention, pick-and-choose whichever is most appropriate for your specific PR:
- [ ] Tag devops on your PR (`@edx/devops`), either for review (more specific to devops = better) or just a general FYI
- [ ] File a support ticket

FYI: Tag anyone who might be interested in this PR here.

## Post-review
- [ ] Squash commits

File: test/acceptance/auto_auth.py [hunk 1]
```diff
@@ -80,27 +80,12 @@ def get_user_id(self):
...
-
-    def get_email(self):
-        """
-        Finds and returns the email address of the current user.
-        """
-        _, email = self._get_username_and_email()
-        return email
-
-    def _get_username_and_email(self):
+    def get_username_and_email(self):
```
Comment (efischer19) [inline test/acceptance/auto_auth.py]: This block was changed in the interest of preventing extraneous bok choy queries (which ended up not solving all our speed problems, but seems reasonable to do anyways)

File: test/acceptance/tests.py [hunk 2]
```diff
@@ -15,8 +16,10 @@
     SubmissionPage, AssessmentPage, GradePage, StaffAreaPage
 )
 
+# This value is generally used in jenkins, but not locally
+PROFILING_ENABLED = os.environ.get('ORA_PROFILING_ENABLED', False)
 
-def retry(tries=4, delay=3, backoff=2):
+def retry(tries=2, delay=4, backoff=2):
```
Comment (efischer19) [inline test/acceptance/tests.py]: Some our biggest tests are more than 3-4 minutes on jenkins, which means trying them more than 2x leads to hitting the SauceLabs imposed 10-minute timeout failure. I think having 2 tries as a default is a good balance between letting one sporadic failure crash a test run, and holding everything up in the case of legitimate errors.

File: test/acceptance/tests.py [hunk 3]
```diff
@@ -472,12 +488,7 @@ def test_staff_area_buttons(self):
 
     @retry()
     @attr('acceptance')
-    @ddt.data(
-        ("staff-tools", "MANAGE INDIVIDUAL LEARNERS"),
-        ("staff-info", "VIEW ASSIGNMENT STATISTICS"),
-    )
-    @ddt.unpack
-    def test_staff_area_panel(self, panel_name, button_label):
```
Comment (efischer19) [inline test/acceptance/tests.py]: By moving these from a ddt test into the test as a loop, we save the extra `auto_auth_page.visit()`. I can revert this part if anyone feels strongly about it - I think it makes sense to test them together.

**Comment by efischer19** [comment]
I've squashed everything into 1 commit for merge-to-master consideration. For posterity, here's a few reverted commits containing some data I gathered during this investigation.

- Results of jenkins run: https://github.com/edx/edx-ora2/commit/8607196642453e39171e473662c2af8b2c42f1af
- Results of local run: https://github.com/edx/edx-ora2/commit/3d37a807bc4f1601a9fb318adb275ca17ef9ac42
- Socket times for both: https://github.com/edx/edx-ora2/commit/3446e36c37b1da4fc0e55bf98a678484007eb76b

@dianakhuang @benpatterson, can I get a review on these changes for merging? They won't solve all our acceptance test problems, but I think our test code would be improved with these changes.

File: test/acceptance/tests.py (hunk 2 above)
Comment (benpatterson) [inline test/acceptance/tests.py]: Not for this PR, but do you think flaky would be a better way to go here?

**Comment by benpatterson** [comment]
:+1: from me

File: test/acceptance/tests.py (hunk 2 above)
Comment (efischer19) [inline test/acceptance/tests.py]: Oh yeah, good point, that definitely sounds do-able. I'll write up a ticket to have the team discuss that possibility.

**Comment by dianakhuang** [comment]
:+1: from me, too.

File: test/acceptance/tests.py (hunk 2 above)
Comment (benpatterson) [inline test/acceptance/tests.py]: solid. thanks @efischer19 . I believe flaky arrived in our consciences after ora2 was built, so it makes sense that we have `retry` here for now.

Comment (efischer19) [inline test/acceptance/tests.py]: Ah, that explains it then, cool. Thanks for all your help here @benpatterson!

https://openedx.atlassian.net/browse/TNL-4143

**Comment by efischer19** [comment]
This branch was passing all Travis checks before I squashed, and there are going to be several Travis builds of master today (as we try to get a release out), so I'm going to go ahead and merge this without a green check.

**Pull request merged**